### PR TITLE
tests: add coverage to feature_addrman.py

### PR DESCRIPTION
**Pull request by kevkevinpal**
I added two new tests that will cover the nNew and nTried tests which add coverage to the if block by checking values larger than our range since we only check for negative values now

adding coverage to these lines
https://github.com/bitcoin/bitcoin/blob/master/src/addrman.cpp#L273
https://github.com/bitcoin/bitcoin/blob/master/src/addrman.cpp#L280

our test seem to only cover the `nTried < 0` and `nNew < 0` scenarios

### CODE REVIEW

**Review by ismaelsadeeq — COMMENTED** [review]
Concept ACK on adding test coverage to `feature_addrman`

You could make it more standard by calculating the maximum `len_new` and `len_tried` with the real values of `ADDRMAN_NEW_BUCKET_COUNT`, `ADDRMAN_TRIED_BUCKET_COUNT`, `ADDRMAN_BUCKET_SIZE`.
This reduced magic numbers and make it easier to update if any of the constants were to be updated in a future PR.

<details>
  <summary>you can calculate them in a first commit</summary>
  
  ```diff
  diff --git a/test/functional/feature_addrman.py b/test/functional/feature_addrman.py
  index 7877f9d302..191a68d612 100755
  --- a/test/functional/feature_addrman.py
  +++ b/test/functional/feature_addrman.py
  @@ -14,7 +14,9 @@ from test_framework.test_framework import BitcoinTestFramework
   from test_framework.test_node import ErrorMatch
   from test_framework.util import assert_equal
   
   -
  +ADDRMAN_NEW_BUCKET_COUNT = 1 << 10
  +ADDRMAN_TRIED_BUCKET_COUNT = 1 << 8
  +ADDRMAN_BUCKET_SIZE = 1 << 6
   def serialize_addrman(
       *,
       format=1,
  @@ -117,17 +119,19 @@ class AddrmanTest(BitcoinTestFramework):
   
           self.log.info("Check that corrupt addrman cannot be read (len_tried)")
           self.stop_node(0)
  +        max_len_tried = ADDRMAN_TRIED_BUCKET_COUNT * ADDRMAN_BUCKET_SIZE
           write_addrman(peers_dat, len_tried=-1)
           self.nodes[0].assert_start_raises_init_error(
  -            expected_msg=init_error("Corrupt AddrMan serialization: nTried=-1, should be in \\[0, 16384\\]:.*"),
  +            expected_msg=init_error(f"Corrupt AddrMan serialization: nTried=-1, should be in \\[0, {max_len_tried}\\]:.*"),
               match=ErrorMatch.FULL_REGEX,
           )
   
           self.log.info("Check that corrupt addrman cannot be read (len_new)")
           self.stop_node(0)
  +        max_len_n_new = ADDRMAN_NEW_BUCKET_COUNT * ADDRMAN_BUCKET_SIZE
           write_addrman(peers_dat, len_new=-1)
           self.nodes[0].assert_start_raises_init_error(
  -            expected_msg=init_error("Corrupt AddrMan serialization: nNew=-1, should be in \\[0, 65536\\]:.*"),
  +            expected_msg=init_error(f"Corrupt AddrMan serialization: nNew=-1, should be in \\[0, {max_len_n_new}\\]:.*"),
               match=ErrorMatch.FULL_REGEX,
           )
   ```
</details>


<details>
   <summary>then use them for the test coverage in second commit</summary>
 
  ```diff
diff --git a/test/functional/feature_addrman.py b/test/functional/feature_addrman.py
  index 191a68d612..eb7eaaf9df 100755
  --- a/test/functional/feature_addrman.py
  +++ b/test/functional/feature_addrman.py
  @@ -126,6 +126,13 @@ class AddrmanTest(BitcoinTestFramework):
               match=ErrorMatch.FULL_REGEX,
           )
   
  +        self.log.info("Check that corrupt addrman cannot be read (large len_tried)")
  +        write_addrman(peers_dat, len_tried=max_len_tried + 1)
  +        self.nodes[0].assert_start_raises_init_error(
  +            expected_msg=init_error(f"Corrupt AddrMan serialization: nTried={max_len_tried + 1}, should be in \\[0, {max_len_tried}\\]:.*"),
  +            match=ErrorMatch.FULL_REGEX,
  +        )
  +
           self.log.info("Check that corrupt addrman cannot be read (len_new)")
           self.stop_node(0)
           max_len_n_new = ADDRMAN_NEW_BUCKET_COUNT * ADDRMAN_BUCKET_SIZE
  @@ -135,6 +142,14 @@ class AddrmanTest(BitcoinTestFramework):
               match=ErrorMatch.FULL_REGEX,
           )
   
  +        self.log.info("Check that corrupt addrman cannot be read (large len_new)")
  +        self.stop_node(0)
  +        write_addrman(peers_dat, len_new=max_len_n_new + 1)
  +        self.nodes[0].assert_start_raises_init_error(
  +            expected_msg=init_error(f"Corrupt AddrMan serialization: nNew={max_len_n_new + 1}, should be in \\[0, {max_len_n_new}\\]:.*"),
  +            match=ErrorMatch.FULL_REGEX,
  +        )
  +
```
  </details>
  
  see https://github.com/bitcoin/bitcoin/commit/e309230694512b03c055409f5d70136ab51719f6 and https://github.com/bitcoin/bitcoin/commit/19921ce5e95624469e64d00bf9194b4a67c1ce95

**Comment by kevkevinpal** [comment]
> Concept ACK on adding test coverage to `feature_addrman`
> 
> You could make it more standard by calculating the maximum `len_new` and `len_tried` with the real values of `ADDRMAN_NEW_BUCKET_COUNT`, `ADDRMAN_TRIED_BUCKET_COUNT`, `ADDRMAN_BUCKET_SIZE`. This reduced magic numbers and make it easier to update if any of the constants were to be updated in a future PR.
> you can calculate them in a first commit
> then use them for the test coverage in second commit
> 
> see [e309230](https://github.com/bitcoin/bitcoin/commit/e309230694512b03c055409f5d70136ab51719f6) and [19921ce](https://github.com/bitcoin/bitcoin/commit/19921ce5e95624469e64d00bf9194b4a67c1ce95)

Makes sense took your two commits https://github.com/bitcoin/bitcoin/commit/e309230694512b03c055409f5d70136ab51719f6 and https://github.com/bitcoin/bitcoin/commit/19921ce5e95624469e64d00bf9194b4a67c1ce95 squashed them into one and added you as coauthor thanks for the review!

File: test/functional/feature_addrman.py [hunk 1]
```diff
@@ -14,7 +14,9 @@
 from test_framework.test_node import ErrorMatch
 from test_framework.util import assert_equal
 
-
+ADDRMAN_NEW_BUCKET_COUNT = 1 << 10
+ADDRMAN_TRIED_BUCKET_COUNT = 1 << 8
+ADDRMAN_BUCKET_SIZE = 1 << 6
```
Comment (ismaelsadeeq) [inline test/functional/feature_addrman.py:19]: nit:
space between constant declarations and functions

**Comment by kevkevinpal** [comment]
> CI failure is unrelated?

yea I think so looks like the test is failing in `test/functional/wallet_fundrawtransaction.py`

File: test/functional/feature_addrman.py (hunk 1 above)
Comment (kevkevinpal) [reply test/functional/feature_addrman.py:19]: pushed fix in [eb195d6](https://github.com/bitcoin/bitcoin/pull/28176/commits/eb195d66bcc1779b1385b5c0cd2e28dd081c1f0a)

Comment (stratospher) [reply test/functional/feature_addrman.py:19]: maybe we could move these globals into a common file? it's used in `test/functional/rpc_net.py` in #28523 too. (cc @0xB10C)

File: test/functional/feature_addrman.py [hunk 2]
```diff
@@ -117,17 +120,34 @@ def run_test(self):
...
+        self.log.info("Check that corrupt addrman cannot be read (large len_tried)")
+        write_addrman(peers_dat, len_tried=max_len_tried + 1)
+        self.nodes[0].assert_start_raises_init_error(
+            expected_msg=init_error(f"Corrupt AddrMan serialization: nTried={max_len_tried + 1}, should be in \\[0, {max_len_tried}\\]:.*"),
             match=ErrorMatch.FULL_REGEX,
         )
 
         self.log.info("Check that corrupt addrman cannot be read (len_new)")
         self.stop_node(0)
+        max_len_n_new = ADDRMAN_NEW_BUCKET_COUNT * ADDRMAN_BUCKET_SIZE
```
Comment (stratospher) [inline test/functional/feature_addrman.py:139]: eb195d6: `s/max_len_n_new/max_len_new` for consistency in naming with `max_len_tried`?

**Review by stratospher — COMMENTED** [review]
ACK eb195d6.

File: test/functional/feature_addrman.py (hunk 2 above)
Comment (kevkevinpal) [reply test/functional/feature_addrman.py]: fixed in [1b044a3](https://github.com/bitcoin/bitcoin/pull/28176/commits/1b044a309d48821ae014256eb56d0f83a1de969b)

File: test/functional/feature_addrman.py (hunk 1 above)
Comment (kevkevinpal) [reply test/functional/feature_addrman.py:19]: > maybe we could move these globals into a common file? it's used in `test/functional/rpc_net.py` in #28523 too. (cc @0xB10C)

the best places I can think to make these values common are `test/functional/test_framework/util.py` or `test/functional/test_framework/netutil.py`

**Review by ismaelsadeeq — APPROVED** [review]
Tested ACK 1b044a309d48821ae014256eb56d0f83a1de969b

**Comment by kevkevinpal** [comment]
force pushed to [d2f87a3](https://github.com/bitcoin/bitcoin/pull/28176/commits/d2f87a3f5bc6ac833fee6e5d05452ebe2e137f22)

This addresses https://github.com/bitcoin/bitcoin/pull/28176#discussion_r1341564031 and moves the constant variables to the netutil.py file (cc @0xB10C)

File: test/functional/test_framework/netutil.py [hunk 3]
```diff
@@ -25,6 +25,10 @@
 STATE_LISTEN = '0A'
 # STATE_CLOSING = '0B'
```
Comment (0xB10C) [inline test/functional/test_framework/netutil.py:27]: nit
```suggestion

# Address manager size constants as defined in addrman_impl.h
```

**Review by 0xB10C — APPROVED** [review]
Code Review ACK d2f87a3f5bc6ac833fee6e5d05452ebe2e137f22

File: test/functional/test_framework/netutil.py (hunk 3 above)
Comment (kevkevinpal) [reply test/functional/test_framework/netutil.py:27]: thanks! resolved in [380130d](https://github.com/bitcoin/bitcoin/pull/28176/commits/380130d9d70f8f8d395949a51f43806f6e600efa)

**Comment by ismaelsadeeq** [comment]
ACK 380130d9d70f8f8d395949a51f43806f6e600efa, code looks good to me 🍃 .
Tests passed this branch rebased on master.

**Comment by 0xB10C** [comment]
Re-ACK 380130d9d70f8f8d395949a51f43806f6e600efa

**Pull request merged**